tests/Filter/ComplexFilterTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Behat Gherkin Parser.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\Behat\Gherkin\Filter;
+
+use Behat\Gherkin\Filter\ComplexFilter;
+use Behat\Gherkin\Node\FeatureNode;
+use Behat\Gherkin\Node\ScenarioInterface;
+use Behat\Gherkin\Node\ScenarioNode;
+
+class ComplexFilterTest extends FilterTestCase
+{
+    public function testFilterFeature(): void
+    {
+        $scenarios = [new ScenarioNode('Scenario#1', [], [], '', 1)];
+        $feature = new FeatureNode('Feature#1', null, [], null, $scenarios, '', '', null, 1);
+
+        $filter = $this->createComplexFilter([$feature], $scenarios);
+
+        $filtered = $filter->filterFeature($feature);
+
+        $this->assertCount(1, $scenarios = $filtered->getScenarios());
+        $this->assertSame('Scenario#1', $scenarios[0]->getTitle());
+    }
```

Comment:
I'm not sure this is really testing much, it could pass just by `isScenarioMatch` returning true? 

So either:
* if you're just trying to prove the interface, the class implemented in `createComplexFilter` is more complicated than it needs to be and should just return `true` from the match methods.
* if you think we need to test the logic of abstract ComplexFilter separately to the classes that extend it, there should be more test cases to exercise the stub class properly (e.g. with no matching features/scenarios, with a subset of matches, etc).

It depends on the motivation for adding this test?

Reply:
> it could pass just by isScenarioMatch returning true?

I was thinking I might want to test other scenarios, but now that I think about it it's overcomplicated for no reason. I'll simplify it into one test (essentially I just want to test `filterFeature()` because it was never covered by tests since classes extending this abstract class seem to always override `filterFeature()`).